### DISCUSSION TITLE
FR: Enable / Disable Drag'n'Drop for Structure Overview

### DISCUSSION THREAD

**Discussion by hiasl** [craftcms/cms#12424] [Ideas]
We are dealing with a large customer with loads of entries within multiple structures (nested).
The customer is afraid of accidentally reordering or nesting entries in a structure.

Suggested solution
- Possibility to enable / disable **reordering** entries via drag'n'drop in structure overviews.
- Possibility to enable / disable **nesting** entries via drag'n'drop in structure overviews.

Suggested ways to do that:
- With a UI option from within the structure (unlocking)
- As config option either generally (or per structure)
- As a right on user or usergroup

This feature should have no influence on setting the parent of an entry from within the edit view.

**Comment by hiasl** [comment]
Still a pretty good idea 😀

**Comment by thupsi** [comment]
I _guess_ this is also an accessibility issue: People with hand tremors are more likely to reorder or nest elements by mistake.

**Reply by hiasl** [reply]
Related to this: https://github.com/craftcms/cms/discussions/16749

**Comment by timkelty** [comment]
I think a nice solution for this could be to introduce the concept of "locking" hierarchy/URIs for a given structure or node in a structure as a user permission.
